grpc-java/grpc-servlet-jakarta/src/main/java/io/grpc/servlet/jakarta/ServletServerStream.java

```diff
@@ -303,8 +303,13 @@ public void cancel(Status status) {
             close(Status.CANCELLED.withCause(status.asRuntimeException()), new Metadata());
             CountDownLatch countDownLatch = new CountDownLatch(1);
             transportState.runOnTransportThread(() -> {
-                asyncCtx.complete();
-                countDownLatch.countDown();
+                try {
+                    asyncCtx.complete();
+                    countDownLatch.countDown();
```

Comment:
i think the countdown latch should go in finally, so that even if complete throws we still unblock the calling thread?